toolbox/repo/validate_role_vars_used.py

```diff
@@ -0,0 +1,78 @@
+#! /usr/bin/env python
+
+import glob, os
+import yaml
+import subprocess
+import pathlib
+
+THIS_DIR = pathlib.Path(__file__).absolute().parent
+
+# go to top directory
+os.chdir(os.path.join(THIS_DIR, "..", ".."))
+
+ROLES_VARS_GLOB = "roles/*/vars/*/*"
+ROLES_DEFAULTS_GLOB = "roles/*/defaults/*/*"
+
+def validate_role_vars_used(filename, yaml_doc):
+    errors = 0
+
+    role_name = filename.split("/", 2)[1]
+    for key in yaml_doc.keys():
+        proc = subprocess.run(["grep", key, f"roles/{role_name}",
+                               "--recursive",
+                               ], capture_output=True)
+
+        if proc.returncode != 0:
+            # grep should always find 'key' inside 'filename'
+            # I couldn't find how to exclude one this file with grep options.
+            print(proc.stderr)
+            print("ERROR: grep shouldn't have failed ...")
```

Comment:
Maybe put the subprocess command arguments list in a variable and include it in this print statement, this way people can copy that command and run it manually to try to better understand what went wrong

Reply:
+1 for printing the command, 
but not sure what you mean with
>  Not just here, might be helpful to print it in case of any error in this role

if I always print it, the output will be very verbose

Reply:
There's another error path - `if count = 0:` - it should be there too

Reply:
if `count == 0` (0 lines of stdout), `grep` return `1`, right?

Reply:
Right